core/src/main/scala/cats/arrow/FunctionK.scala

```diff
@@ -0,0 +1,31 @@
+package cats
+package arrow
+
+import cats.data.{Xor, Coproduct}
+
+trait FunctionK[F[_], G[_]] extends Serializable { self =>
+  def apply[A](fa: F[A]): G[A]
+
+  def compose[E[_]](f: FunctionK[E, F]): FunctionK[E, G] =
+    new FunctionK[E, G] {
+      def apply[A](fa: E[A]): G[A] = self.apply(f(fa))
+    }
+
+  def andThen[H[_]](f: FunctionK[G, H]): FunctionK[F, H] =
+    f.compose(self)
+
+  def or[H[_]](h: FunctionK[H,G]): FunctionK[Coproduct[F, H, ?],G] =
+    new (FunctionK[Coproduct[F, H, ?],G]) {
```

Comment:
The parentheses seem to be a remnant of the use of `~>`  (`new (Coproduct[F, G, ?] ~> H) {`). I will remove them.